src/main/java/org/spongepowered/common/data/manipulator/mutable/entity/SpongeAgeableData.java

```diff
@@ -0,0 +1,129 @@
+import org.spongepowered.common.data.manipulator.immutable.entity.ImmutableSpongeAgeableData;
+import org.spongepowered.common.data.manipulator.mutable.common.AbstractData;
+import org.spongepowered.common.data.value.SpongeValueFactory;
+import org.spongepowered.common.data.value.mutable.SpongeValue;
+
+public class SpongeAgeableData extends AbstractData<AgeableData, ImmutableAgeableData> implements AgeableData {
+
+    private int age;
+    private boolean baby;
+    private boolean adult;
+
+    public SpongeAgeableData(int age, boolean baby, boolean adult) {
+        super(AgeableData.class);
+        this.age = age;
+        this.baby = baby;
+        this.adult = adult;
+        registerGettersAndSetters();
+    }
+
+    @Override
+    public AgeableData copy() {
+        return new SpongeAgeableData(this.age, this.baby, this.adult);
+    }
+
+    @Override
+    public ImmutableAgeableData asImmutable() {
+        return new ImmutableSpongeAgeableData(this.age, this.baby, this.adult);
+    }
+
+    @Override
+    public int compareTo(AgeableData o) {
+        return ComparisonChain.start()
+                .compare(o.age().get().intValue(), this.age)
+                .compare(o.baby().get(), this.baby)
+                .compare(o.adult().get(), this.adult)
+                .result();
+    }
+
+    @Override
+    public MutableBoundedValue<Integer> age() {
+        return SpongeValueFactory.boundedBuilder(Keys.AGE)
+                .minimum(Integer.MIN_VALUE)
+                .maximum(Integer.MAX_VALUE)
+                .defaultValue(0)
+                .actualValue(this.age)
+                .build();
+    }
+
+    @Override
+    public Value<Boolean> baby() {
+        return new SpongeValue<>(Keys.BABY, this.baby);
+    }
+
+    @Override
+    public Value<Boolean> adult() {
+        return new SpongeValue<>(Keys.ADULT, this.adult);
+    }
+
+    @Override
+    protected void registerGettersAndSetters() {
+        registerFieldGetter(Keys.AGE, SpongeAgeableData.this::getAge);
+        registerFieldSetter(Keys.AGE, SpongeAgeableData.this::setAge);
+        registerKeyValue(Keys.AGE, SpongeAgeableData.this::age);
+
+        registerFieldGetter(Keys.BABY, SpongeAgeableData.this::isBaby);
+        registerFieldSetter(Keys.BABY, SpongeAgeableData.this::setBaby);
+        registerKeyValue(Keys.BABY, SpongeAgeableData.this::baby);
+
+        registerFieldGetter(Keys.ADULT, SpongeAgeableData.this::isAdult);
+        registerFieldSetter(Keys.ADULT, SpongeAgeableData.this::setAdult);
+        registerKeyValue(Keys.ADULT, SpongeAgeableData.this::adult);
+    }
+
+    public int getAge() {
+        return age;
+    }
+
+    public void setAge(int age) {
+        this.age = age;
+    }
+
+    public boolean isAdult() {
+        return adult;
+    }
+
+    public void setAdult(boolean adult) {
+        this.adult = adult;
+    }
+
+    public boolean isBaby() {
+        return baby;
+    }
+
+    public void setBaby(boolean baby) {
```

Comment:
Is it really possible to have some random age, being a baby and an adult at the same time?

From the API javadocs I would assume all is somehow bound to the age, at least: 
`adult <=> !baby`.

[AgeableData](https://github.com/SpongePowered/SpongeAPI/blob/master/src/main/java/org/spongepowered/api/data/manipulator/mutable/entity/AgeableData.java#L55)